dockertest.go

```diff
@@ -505,6 +505,37 @@ func (d *Pool) RunWithOptions(opts *RunOptions, hcOpts ...func(*dc.HostConfig))
 	}, nil
 }
 
+// inspectContainerWithRetries will repeat the inspect call until the container has port bindings assigned.
+func (d *Pool) inspectContainerWithRetries(id string) (*dc.Container, error) {
+	const maxRetries = 10
+	var (
+		retryNum int
+		c        *dc.Container
+		err      error
+	)
+	for retryNum <= maxRetries {
+		if retryNum > 0 {
+			time.Sleep(100 * time.Millisecond)
+		}
+		c, err = d.Client.InspectContainer(id)
+		if err != nil {
+			return nil, err
+		}
+		if hasEmptyPortBindings := func() bool {
+			for _, bindings := range c.NetworkSettings.Ports {
+				if len(bindings) == 0 {
+					return true
+				}
+			}
+			return false
+		}(); !hasEmptyPortBindings {
+			return c, nil
+		}
+		retryNum++
```

Comment:
Can be placed into the `for` [`PostStmt`](https://go.dev/ref/spec#For_clause).

Created the PR #544 to show this